### DISCUSSION TITLE
admin enroll students?

### DISCUSSION THREAD

**Discussion by KlavsKlavsen** [inducer/relate#925] [Q&A]
Hi,
How do I enroll users in an exam?
Only way I can find currently - is to guide them to login and click enroll.. 
But I would like to just send them an exam ticket instead - and have that work - but it fails with "user is not enrolled in that course" - when I try to issue one.

I tried to find docs talking about HOW the flow for this is supposed to work - but searching on documentation site is broken (never responds when I search for 'enroll' ) and on github I found no issues that seemed to explain that.

I would like courses to NOT be visible to anyone - except the ones whom I give an exam.
I have so far only gotten this to work, by making courses publicly available - but I will try to follow this https://github.com/inducer/relate/discussions/865 - to make it only visible to those who are enrolled - and then impersonate user and "open the course link because I know of it" - and enroll there..
Currently if I visit the course page, impersonating my new user (and course is set to not accept enrollment - as I would like to enroll them myself) - I get this error (but there is no 'enroll' button shown):
```
 You do not have any existing/viewable sessions and are not allowed to start a new one.

Check the following:

    Has the deadline for this assignment passed?
    Complete your enrollment in your course. If you're not enrolled, a large "Enroll now" button will show up at the top of your course page. Click that, follow the indicated steps, and then retry your last action.
    If none of the above steps help, please navigate back to your course from the home page and contact your course staff.
```

**Comment by inducer** [comment]
There are two (technically three) stages.
- First, people need to have an account. (create those from the admin interface)
- Second, those accounts need to be participants in the course. (add those manually under "Grading > Participants" or allow self-enrollment via "Instructor > Preapprove enrollments")
- Finally, you can issue exam tickets for those participations.

I'd be receptive to something that streamlines the first two steps, given, say, a list of emails.
